src/tools/compiletest/src/runtest.rs

```diff
@@ -295,11 +295,10 @@ impl<'test> TestCx<'test> {
     }
 
     fn check_correct_failure_status(&self, proc_res: &ProcRes) {
-        // The value the rust runtime returns on failure
-        const RUST_ERR: i32 = 101;
-        if proc_res.status.code() != Some(RUST_ERR) {
+        if proc_res.status.success() {
             self.fatal_proc_rec(
-                &format!("failure produced the wrong error: {}", proc_res.status),
+                &format!("failure must not return success exit status!  Returned status: {}",
```

Comment:
If the branch is taken, the error code is always 0, so printing it is not required ;)

Reply:
Ha!  That is an excellent point! :)